File: dataset/output/GEMINI_gemini-1.5-pro-001/Task109_GEMINI_gemini-1.5-pro-001.c
#include <stdio.h>
#include <regex.h>

int main() {
    char email[100];
    regex_t regex;
    int result;

    printf("Enter email address: ");
    fgets(email, sizeof(email), stdin);

    // Remove newline character from fgets()
    email[strcspn(email, "
")] = 0;

    // Compile the regular expression
    result = regcomp(&regex, "^[A-Za-z0-9+_.-]+@(.+)$", REG_EXTENDED);
    if (result) {
        fprintf(stderr, "Could not compile regex
");
        return 1;
    }

    // Match the email against the regex
    result = regexec(&regex, email, 0, NULL, 0);
    if (!result) {
        printf("Valid email address
");
    } else {
        printf("Invalid email address
");
    }

    // Free the compiled regex
    regfree(&regex);

    return 0;
}

File: dataset/output/GEMINI_gemini-1.5-pro-001/Task109_GEMINI_gemini-1.5-pro-001.java
import java.util.Scanner;
import java.util.regex.Pattern;

public class ValidateInput {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter email address: ");
        String email = scanner.nextLine();
        
        String emailRegex = "^[A-Za-z0-9+_.-]+@(.+)$";
        Pattern pattern = Pattern.compile(emailRegex);
        
        if (pattern.matcher(email).matches()) {
            System.out.println("Valid email address");
        } else {
            System.out.println("Invalid email address");
        }
    }
}